controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -300,8 +300,8 @@ func (r *StorageConsumerReconciler) reconcileEnabledPhases() (reconcile.Result,
 		// A provider cluster already has a NooBaa system and does not require a NooBaa account
 		// to connect to a remote cluster, unlike client clusters.
 		// A NooBaa account only needs to be created if the storage consumer is for a client cluster.
-		clusterID := util.GetClusterID(r.ctx, r.Client, &r.Log)
-		if clusterID != "" && !strings.Contains(r.storageConsumer.Name, clusterID) {
+		//TODO: (rewant) This is a hack that we are using and this needs to go away, we shouldn't rely on consumer name
+		if r.storageConsumer.Name != defaults.LocalStorageConsumerName {
```

Comment:
I would prefer if we compare the local cluster id with the one on .Status.Client.ClusterID